### PR TITLE
add cardano-node to shell.nix and integration build inputs

### PR DESCRIPTION
**Pull request by disassembler**
# Issue Number

# Overview

- [x] I have added cardano-node executable to be in-scope for integration tests.

# Comments

### CODE REVIEW

File: nix/pkgs.nix [hunk 1]
```diff
@@ -1,8 +1,11 @@
 # our packages overlay
 let
-  pkgs1903 = import (import ./sources.nix {})."nixpkgs-19.03" {};
+  sources = import ./sources.nix {};
+  pkgs1903 = import sources."nixpkgs-19.03" {};
 in pkgs: _: with pkgs; {
   jmPkgs = import ./jormungandr.nix { inherit (pkgs) commonLib; inherit pkgs; };
+  # TODO: do we need a system/config here somehow???
+  nodePkgs = import sources."cardano-node" {};
```
Comment (jbgi) [inline nix/pkgs.nix]: possibly
```suggestion
  nodePkgs = import sources."cardano-node" { inherit config system; };
```
but not sure how that would play with cross-compilation. Probably the cleanest would be to move this import directly to `nix/default.nix` and then import with all the arguments (system, crossSystem, config and sourcesOverride).

**Comment by rvl** [comment]
I rebased (merge conflicts) and fixed the crossSystem/config issue.

**Review by rvl — APPROVED** [review]

**Comment by rvl** [comment]
It's almost ready but ouroboros-network (dependency of cardano-node) is failing to build for Windows.
Fixed by PR input-output-hk/cardano-node#641.

**Comment by rvl** [comment]
We also need to patch http-client-0.6.4 because that is failing to build on Windows:
https://hydra.iohk.io/job/Cardano/cardano-node/x86_64-w64-mingw32.cardano-node.x86_64-linux

PR input-output-hk/cardano-node#646

**Comment by rvl** [comment]
I will merge this PR now and open another PR once the windows build of cardano-node is fixed.

bors r+

**Comment by rvl** [comment]
The buildkite agents were having ZFS issues. Trying again.

bors r+

**Comment by rvl** [comment]
The tests should succeed now that #1420 is merged.

bors r+

**Pull request merged**